spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java

```diff
@@ -470,6 +470,28 @@ public void testMessagingGateway() throws InterruptedException {
 		assertThat(this.asyncAnnotationProcessThread.get()).isNotSameAs(Thread.currentThread());
 	}
 
+	/**
+	 * Just creates an interim context to confirm that the DefaultConfiguringBeanFactoryPostProcessor 
+	 * does not fail when there is an extra application context in the hierarchy.
+	 */
+	@Test
+	public void testDoubleParentChildAnnotationConfiguration() {
+		AnnotationConfigApplicationContext parent;
+		parent = new AnnotationConfigApplicationContext();
+		parent.register(ChildConfiguration.class);
+		parent.setParent(this.context);
+		parent.refresh();
+
+		AnnotationConfigApplicationContext child;
+		child = new AnnotationConfigApplicationContext();
+		child.register(ChildConfiguration.class);
+		child.setParent(parent);
+		child.refresh();
+
+		parent.close();
+		child.close();
```

Comment:
This test doesn't verify anything. It is just not a test.
This is indeed not enough just to say it should not fail. It even won't fail for some other reasons we may make changes.
So, if we add a test it must verify something.

Why don't check that child context doesn't create a new `NullChannel` just because there is one in the parent?
More over I think such a test won't fail even in the current state of code. just because your original concern that we don't go deep into the hierarchy. So, a test must have at least 3 "relatives" in the parent-child relationship.

Reply:
Fair point though there are 3 levels as the default level coming from the test itself is one of the levels.
This test will definitely fail on the current code.

But I'll see what verify I can add - check that a null channel is not added to the child context would be something.

Reply:
Assertions added as suggested. This does make a lot of sense - provies the null channel is being found in each case by the parent context.